src/theory/partition_generator.cpp

```diff
@@ -141,6 +159,57 @@ TrustNode PartitionGenerator::makeRevisedPartitions()
   }
 }
 
+
+TrustNode PartitionGenerator::makeFullTrailPartitions()
+{
+  std::vector<TNode> literals = collectDecisionLiterals();
+  uint64_t num_var = static_cast<uint64_t>(log2(d_numPartitions));
+  if (literals.size() >= num_var)
+  {
+    literals.resize(num_var);
+    std::vector<TNode> part_nodes;
+
+    // This complicated thing is basically making a truth table
+    // of with 2^num_var variable so that these can be put together emitted as a partition
+    // later. Each entry in result_node_lists is a row corresponding to a cube:
+    // result_node_lists = {
+    //   { l1,  l2}
+    //   { l1, !l2}
+    //   {!l1,  l2}
+    //   {!l1, !l2} }
+    // result_node_lists is built column by column. 
+    std::vector<std::vector<TNode> > result_node_lists(pow(2, num_var));
+    bool t = false;
+    size_t q = num_var;
+    for (TNode n : literals)
+    {
+      TNode not_n = n.notNode();
```

Comment:
This shouldn't be a `TNode` because `notNode()` creates a new node and it must be kept alive until it is used.